_posts/2016-05-22-th1eadin0g1.markdown

```diff
@@ -0,0 +1,226 @@
+---
+layout: post
+title: "Threading 101 for iOS"
+date: 2016-05-22 12:09
+author: orta
+categories: [mobile, cocoapods, newbies]
+---
+
+Once an app has reached a level of complexity wherein Apple would _not_ [reject it](https://developer.apple.com/app-store/review/rejections/) for "Not enough lasting value" it has probably had to deal with multiple thread execution.
+
+This could be as simple as downloading a file, or image from the internet, accessing an external API, using something like background locations or playing sounds. There are no limits to the amount of cases where you could be doing some asynchronous work under the hood at the same time. So, I'm going to try and cover _most_ of the common use cases of where you are using it without knowing, then expand on a few techniques to write useful multi-threaded code.
+
+<!-- more -->
+
+### Thread Terminology 101
+
+* `Process` - The context within an app is running, you will have one process per app. 
+* `Thread` - A pathway of code, you can have multiple threads in an app's process.
+* `Main Thread` - The initial thread for your app, this is the only thread where you can do user interface code.
```

Comment:
I would say "also known as a UI thread, because all UI updates are required to happen on a single thread" or something to that effect.